letas_repetidas.rb

```diff
@@ -0,0 +1,41 @@
+def repetidas(firstWord, secondWord)
+    firstArrWords = firstWord.chars
+    secondArrWords = secondWord.chars
+    repetidasArr = []
+    if firstArrWords.length <= secondArrWords.length
+        getDuplicates(firstArrWords, secondArrWords, repetidasArr)
+    else
+        getDuplicates(secondArrWords, firstArrWords, repetidasArr)        
+    end    
+
+    if(repetidasArr.length > 0)
+        
+        repetidasArrSort = repetidasArr.sort
+        letrasRepetidas =""
+        repetidasArrSort.each do |v|
+            letrasRepetidas += v +", "
+        end        
+        puts "Letras: #{letrasRepetidas}"
+        puts "Total letras: #{repetidasArrSort.length}"
+    end
+end
+
+def getDuplicates(firstArr, secondArr, repetidasArr)
+    firstArr.each do |v|
+        if !v !="" && secondArr.include?(v)
+            if repetidasArr.select{|w| w == v}.length == 0
+                repetidasArr.push(v)
+            end
+            
+        end
+    end
+end
+
+
+puts "Bienvenido al programa, Detectar letras repetidas"
+puts "Ingresa la primera palabra."
+firstWord = gets.chomp
+puts "Ingresa la segunda palabra."
+secondWord = gets.chomp
+puts repetidas(firstWord, secondWord)
+
```

Comment:
check this enhancement:
```suggestion
def words_repeted(words)
  first_word = words.first.chars
  last_word = words.last.chars

  repeted_words = get_duplicated_words(first_word, last_word)

  unless (repeted_words.empty?)
    puts "Letras: #{repeted_words.sort.join(', ')}"
    puts "Total letras: #{repeted_words.length}"
  end
end

def get_duplicated_words(first_word, second_word)
  first_word.select { |letter| second_word.include?(letter) }.uniq!
end


puts 'Bienvenido al programa, Detectar letras repetidas'
puts 'Ingresa dos palabras'

words = gets.chomp.split(' ')

puts words_repeted(words)
```